pkg/build/builder/sti.go

```diff
@@ -92,9 +92,16 @@ func (s *S2IBuilder) Build() error {
 	if s.build.Spec.Strategy.SourceStrategy == nil {
 		return fmt.Errorf("the source to image builder must be used with the source strategy")
 	}
-
+	var runtimeImageName string
 	var push bool
 
+	runtimeImage := s.build.Spec.Strategy.SourceStrategy.RuntimeImage
```

Comment:
at some point you need to resolve the RuntimeImage reference to a docker pull spec, because the ObjectReference could be to an ImageStreamTag (name+namespace point to an imagestreamtag from which you can get a pullspec), or a DockerImage (name is the docker pull spec to use).

See what we do with buildImage, this will be the same:
https://github.com/openshift/origin/blob/master/pkg/build/generator/generator.go#L482-L504